src/python/library/tritonclient/grpc/__init__.py

```diff
@@ -1699,11 +1699,19 @@ def set_data_from_numpy(self, input_tensor):
         """
         if not isinstance(input_tensor, (np.ndarray,)):
             raise_error("input_tensor must be a numpy array")
-        dtype = np_to_triton_dtype(input_tensor.dtype)
-        if self._input.datatype != dtype:
-            raise_error(
-                "got unexpected datatype {} from numpy array, expected {}".
-                format(dtype, self._input.datatype))
+        # DLIS-3720: Special handling for bfloat16 until Numpy officially supports it
```

Comment:
>DLIS-3720

This should changed to be the follow up ticket which will remove this special handling when python supports BF16

Reply:
Thanks, done.